File: py/abc_011/c.py
n = int(input())
NG1 = int(input())
NG2 = int(input())
NG3 = int(input())
arr = sorted([NG1,NG2,NG3], reverse=True)
i = 0
count = 0
tmp = n
flg = False
if not n in arr:
  for _ in range(100):
    if tmp-3 in arr:
      if tmp-2 in arr:
        if tmp-1 in arr:
          flg = False
          break
        else:
          tmp -= 1
      else:
        tmp -= 2
    else:
      tmp -= 3
    count+=1
    if tmp <= 0:
      flg = True
      break

if flg:
  print('YES')
else:
  print('NO')

File: c++/abc_011/c.cpp
#include <bits/stdc++.h>
using namespace std;

int main () {
  int N;
  cin >> N;

  vector<int> v;
  int tmp;

  for (int i = 0; i < 3; i++) {
    cin >> tmp;
    v.push_back(tmp);
  }

  int flg = false;

  if (*find(v.begin(), v.end(), N) == 0) {
    for (int i = 0; i < 100; i++) {
      if (*find(v.begin(), v.end(), N - 3) == 0 && (N - 3) >= 0) {
        N -= 3;
      } else if (*find(v.begin(), v.end(), N - 2) == 0 && (N - 2) >= 0) {
        N -= 2;
      } else if (*find(v.begin(), v.end(), N - 1) == 0 && (N - 1) >= 0) {
        N -= 1;
      }
      if (N == 0) {
        flg = true;
        break;
      }
    }
  }
  
  if (flg) {
    cout << "YES" << endl;
  } else {
    cout << "NO" << endl;
  }
}